rclcpp/include/rclcpp/node.hpp

```diff
@@ -972,7 +972,11 @@ class Node : public std::enable_shared_from_this<Node>
 
   /// Return a list of parameters with any of the given prefixes, up to the given depth.
   /**
-   * \todo: properly document and test this method.
+   * \param[in] prefixes The list of prefixes that should be searched for within the
+   * current parameters.
```

Comment:
1) Yea I did realize that when looking at function and the tests. I can definitely talk about empty, 0 in both the parameters. Since you can also pass string based parameters should we add those to the `list_parameters` tests, or are empty vectors fine enough? 

2) I will also implement an explanation of prefixes and their separation

Reply:
I was looking at the,

https://github.com/ros2/rclcpp/blob/9284d7cefa9e37ce3b6926a44e6177d6e4a4b62a/rclcpp/test/rclcpp/node_interfaces/test_node_parameters.cpp#L75

probably test cases are not really enough to cover the cases that @clalancette mentioned here. i would recommend the test cases for those as well.

I think we need to explain that `prefix` can match with parameter_name as well.

Reply:
> 1. Since you can also pass string based parameters should we add those to the `list_parameters` tests, or are empty vectors fine enough?

I think we already have those tests in https://github.com/ros2/rclcpp/blob/9284d7cefa9e37ce3b6926a44e6177d6e4a4b62a/rclcpp/test/rclcpp/node_interfaces/test_node_parameters.cpp#L75-L119 .  Unless you mean something else?

Reply:
> i was thinking list_parameters(prefixes, 0u) case that missing here?
That was what I meant, so I added just that test.